website/views.py

```diff
@@ -4744,3 +4746,92 @@ def blt_tomato(request):
         project["funding_hyperlinks"] = funding_link
 
     return render(request, "blt_tomato.html", {"projects": data})
+
+
+@login_required
+def vote_suggestions(request):
+    if request.method == "POST":
+        user = request.user.username
+        data = json.loads(request.body)
+        suggestion_id = data.get("suggestion_id")
+        suggestion = Suggestion.objects.get(id=suggestion_id)
+        up_vote = data.get("up_vote")
+        down_vote = data.get("down_vote")
+        liked = SuggestionLikes.objects.filter(user=user, suggestion=suggestion).exists()
+        if not liked:
+            if up_vote is True:
+                liked = SuggestionLikes.objects.create(
+                    user=user, suggestion=suggestion, up_vote=True, down_vote=False
+                )
+                suggestion.up_vote += 1
+            elif down_vote is True:
+                liked = SuggestionLikes.objects.create(
+                    user=user, suggestion=suggestion, up_vote=False, down_vote=True
+                )
+                suggestion.down_vote += 1
+        else:
+            if up_vote is False:
+                suggestion.up_vote -= 1
+                liked = SuggestionLikes.objects.filter(user=user, suggestion=suggestion).delete()
+                if down_vote is True:
+                    liked = SuggestionLikes.objects.create(
+                        user=user, suggestion=suggestion, down_vote=True, up_vote=False
+                    )
+                    suggestion.down_vote += 1
+            elif down_vote is False:
+                suggestion.down_vote -= 1
+                liked = SuggestionLikes.objects.filter(user=user, suggestion=suggestion).delete()
+                if up_vote is True:
+                    liked = SuggestionLikes.objects.create(
+                        user=user, suggestion=suggestion, up_vote=True, down_vote=False
+                    )
+                    suggestion.up_vote += 1
```

Comment:
Could you simplify this using [update_or_create](https://docs.djangoproject.com/en/5.0/ref/models/querysets/#update-or-create)?

Reply:
will look into it

Reply:
Are you still looking into it? The code seems to be the same

Reply:
Yes , I am working on it !